attack_range.conf.template

```diff
@@ -25,6 +25,10 @@ ip_whitelist = 0.0.0.0/0
 private_key_path = ~/.ssh/id_rsa
 # Specify the path to your private SSH key
 
+use_ssh_agent = 1
+# Specify whether to load the key directly or via agent (required if encrypted)
+# possible values: 1, 0
+
```

Comment:
if the execution will fail if the private key is password-less and this is set to 1, then I would prefer this is set to 0 by default (current behavior). And only if this should be set to 1 if user is using password protected key.

Maybe the text should be more clear in when would you want to set this to 1.

Reply:
It should be set to `1` if you use an agent to handle your ssh key.

It will fail if you set it to `1` and don't use an agent or if you set it to `0` and your key is password-protected.

Reply:
Side note: this only affects terraform. Somehow (maybe because it uses the openssh client directly) ansible is able to figure things out for itself. Since terraform only invokes this remote-exec connection in order to `echo booted`, maybe we can just skip that. I'm not actually sure if that step is functional (i.e. wait-for-boot) or just informational.

Reply:
Looked it up and `echo booted` probably necessary to make sure the ansible step doesn't just fail because the machine is still coming up.

So...

If you guys dislike the additional config, we can make it optional and autodetect whether or not the specified key is managed by an agent. Fortunately paramiko is already a requirement, but unfortunately it's still super hacky because paramiko throws away the key path info.  Here's the kind of hack we're talking about:

```
--- a/modules/CustomConfigParser.py
+++ b/modules/CustomConfigParser.py
@@ -44,6 +44,23 @@ class CustomConfigParser:
                 print("ERROR - in configuration file: {0}, cannot include whitespace in BOTS data set config directive.".format(CONFIG_PATH))
                 sys.exit(1)

+        # Autodetect use_ssh_agent if unspecified
+        if 'use_ssh_agent' not in self.settings:
+            import os.path
+            import paramiko.agent
+            agent = paramiko.agent.Agent()
+            agent_key_paths = set()
+            ptype, result = agent._send_message(paramiko.agent.cSSH2_AGENTC_REQUEST_IDENTITIES)
+            if ptype == paramiko.agent.SSH2_AGENT_IDENTITIES_ANSWER:
+                for i in range(result.get_int()):
+                    result.get_binary()
+                    agent_key_paths.add(result.get_string().decode())
+            key_path = os.path.expanduser(self.settings['private_key_path'])
+            if key_path in agent_key_paths:
+                self.settings['use_ssh_agent'] = "1"
+            else:
+                self.settings['use_ssh_agent'] = "0"
+
```

Reply:
I like the idea of auto-detecting in CustomConfigParser.py.
Regarding the agent, I think ansible has 3 modes, `smart`, `openssh`, `paramiko`. Default is `smart`. Perhaps since ansible is smarter that's why it doesn't complain 😂